src/SimpleThings/EntityAudit/AuditConfiguration.php

```diff
@@ -124,14 +158,25 @@ public function createMetadataFactory()
         return new Metadata\MetadataFactory($this->auditedEntityClasses);
     }
 
-    public function setCurrentUsername($username)
+    public function setUsernameCallable($usernameCallable)
     {
-        $this->currentUsername = $username;
+        // php 5.3 compat
+        if (null !== $usernameCallable && !is_callable($usernameCallable)) {
+            throw new \InvalidArgumentException(sprintf(
+                'Username Callable must be callable. Got: %s',
+                is_object($usernameCallable) ? get_class($usernameCallable) : gettype($usernameCallable)
+            ));
+        }
+
+        $this->usernameCallable = $usernameCallable;
```

Comment:
Suggestion: `$usernameCallable ?: function () { return $this->currentUsername; }`